tensorboard/plugins/histogram/histograms_plugin.py

```diff
@@ -196,11 +212,30 @@ def histograms_impl(self, tag, run, downsample_to=None):
         raise ValueError('No histogram tag %r for run %r' % (tag, run))
       if downsample_to is not None and len(tensor_events) > downsample_to:
         rand_indices = random.Random(0).sample(
-            six.moves.xrange(len(tensor_events)), downsample_to)
+          six.moves.xrange(len(tensor_events)), downsample_to)
         indices = sorted(rand_indices)
         tensor_events = [tensor_events[i] for i in indices]
       events = [[e.wall_time, e.step, tensor_util.make_ndarray(e.tensor_proto).tolist()]
                 for e in tensor_events]
+    if output_format == OutputFormat.CSV:
+      csv_events = []
+      # Convert the events in a way that we can sensibly export
+      # them as csv. Therefore, we split the start, end, value of
+      # each bin by semicolon.
+      for e in events:
+        csv_events.append(
+          [e[0], e[1],
+           ';'.join(['{:.32f}'.format(el[0]) for el in e[2]]),
```

Comment:
Any reason you're using 32 digits of precision?  That seems like overkill; for 64-bit floats, 17 decimal digits of precision should be enough to have a lossless roundtrip conversion.